zio-query/shared/src/main/scala/zio/query/ZQuery.scala

```diff
@@ -92,19 +93,49 @@ final class ZQuery[-R, +E, +A] private (private val step: ZIO[(R, QueryContext),
   final def >>=[R1 <: R, E1 >: E, B](f: A => ZQuery[R1, E1, B]): ZQuery[R1, E1, B] =
     flatMap(f)
 
+  /**
+   * Returns a query which submerges the error case of `Either` into the error channel of the query
+   *
+   * The inverse of [[ZQuery.either]]
+   */
+  def absolve[E1 >: E, B](implicit ev: A <:< Either[E1, B]): ZQuery[R, E1, B] =
+    ZQuery.absolve(self.map(ev))
+
   /**
    * Maps the success value of this query to the specified constant value.
    */
   final def as[B](b: => B): ZQuery[R, E, B] =
     map(_ => b)
 
+  /**
+   * Lifts the error channel into a `Some` value for composition with other optional queries
+   *
+   * @see [[ZQuery.some]]
+   */
+  def asSomeError: ZQuery[R, Option[E], A] =
+    mapError(Some(_))
+
   /**
    * Returns a query whose failure and success channels have been mapped by the
    * specified pair of functions, `f` and `g`.
    */
   final def bimap[E1, B](f: E => E1, g: A => B)(implicit ev: CanFail[E]): ZQuery[R, E1, B] =
     foldM(e => ZQuery.fail(f(e)), a => ZQuery.succeed(g(a)))
 
+  /**
+   * Recovers from all errors.
+   */
+  def catchAll[R1 <: R, E2, A1 >: A](h: E => ZQuery[R1, E2, A1])(implicit ev: CanFail[E]): ZQuery[R1, E2, A1] =
+    self.foldM[R1, E2, A1](h, ZQuery.succeed(_))
+
+  /**
+   * Recovers from all errors with provided Cause.
+   *
+   * @see [[ZQuery.sandbox]] - other functions that can recover from defects
+   */
+  def catchAllCause[R1 <: R, E2, A1 >: A](h: Cause[E] => ZQuery[R1, E2, A1]): ZQuery[R1, E2, A1] =
+    self.foldCauseM[R1, E2, A1](h, ZQuery.succeed(_))
```

Comment:
I notably didn't add all of the variants that exist on `ZIO` this was partly because the carry over wasn't one-to-one and the `ZIO` variants had some custom optimizations related to stack traces that I didn't know how to translate into `ZQuery`